bee-common/bee-common/src/auth/jwt.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides JSON Web Token utilities.
+
+use jsonwebtoken::{decode, encode, errors, DecodingKey, EncodingKey, Header, Validation};
+use serde::{Deserialize, Serialize};
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+/// Error occurring when creating JWTs.
+pub type Error = errors::Error;
+
+/// Represents registered JSON Web Token Claims.
+/// https://tools.ietf.org/html/rfc7519#section-4.1
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Claims {
```

Comment:
How about making `Claims` lifetime-generic, and then use `&'a str` or `Cow<'a, str>` instead of `String`.

Reply:
I opened an issue for later, don't really have time to investigate consequences of that in the codebase.
https://github.com/iotaledger/bee/issues/504

Reply:
alright :+1: